src/Content/Text/BBCode.php

```diff
@@ -1978,4 +1982,148 @@ private static function getAbstract($text, $addon = "")
 
 		return $abstract;
 	}
+
+	/**
+	 * @brief Callback function to replace a Friendica style mention in a mention for Diaspora
+	 *
+	 * @param array $match Matching values for the callback
+	 * @return string Replaced mention
+	 */
+	private static function bbCodeMention2DiasporaCallback($match)
+	{
+		$contact = Contact::getDetailsByURL($match[3]);
+
+		if (empty($contact['addr'])) {
+			$contact = Probe::uri($match[3]);
+		}
+
+		if (empty($contact['addr'])) {
+			return $match[0];
+		}
+
+		$mention = '@{' . $match[2] . '; ' . $contact['addr'] . '}';
+		return $mention;
+	}
+
+	/**
+	 * @brief Converts a BBCode text into Markdown
+	 *
+	 * This function converts a BBCode item body to be sent to Markdown-enabled
+	 * systems like Diaspora and Libertree
+	 *
+	 * @param string $text
+	 * @param bool   $for_diaspora Diaspora requires more changes than Libertree
+	 * @return string
+	 */
+	public static function toMarkdown($text, $for_diaspora = true)
+	{
+		$a = self::getApp();
+
+		$original_text = $text;
+
+		// Since Diaspora is creating a summary for links, this function removes them before posting
+		if ($for_diaspora) {
+			$text = self::removeShareInformation($text);
+		}
+
+		/**
```

Comment:
please don't use `/**` for comments because `/**` would add the comment to the doxygen docu.

Reply:
I didn't write it but I could have fixed it indeed.